HerPublicWebsite.BusinessLogic/QuestionnaireUpdater.cs

```diff
@@ -125,6 +126,17 @@ public async Task<Questionnaire> GenerateReferralAsync(Questionnaire questionnai
             );
         }
 
+        var perReferralReport = new PerReferralReport(referralRequest);
+        await dataAccessProvider.PersistPerReferralReportAsync(perReferralReport);

```

Comment:
Please put a try/catch around this and log an error in the catch. If this goes wrong for any reason we want to know about it, but it shouldn't stop the user of the site continuing with their journey